src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -566,7 +613,7 @@ internal sealed override ObsoleteAttributeData ObsoleteAttributeData
         public abstract bool HasReferenceTypeConstraint { get; }
 
         // PROTOTYPE(NullableReferenceTypes): Should this API be exposed through ITypeParameterSymbol?
-        internal abstract bool HasNullableReferenceTypeConstraint { get; }
+        internal abstract bool? ReferenceTypeConstraintIsNullable { get; }
```

Comment:
>ReferenceTypeConstraintIsNullable [](start = 32, length = 33)

xml doc on this would be useful (what does `null` return mean?)

Reply:
> xml doc on this would be useful (what does null return mean?)

I think it means what null always mean - unknown

---
In reply to: [213858940](https://github.com/dotnet/roslyn/pull/29586#discussion_r213858940) [](ancestors = 213858940)

Reply:
What do you get when there is no `class` or `class?`? Is there a difference between `class` in NNT(true) or NNT(false) context?

---
In reply to: [213859517](https://github.com/dotnet/roslyn/pull/29586#discussion_r213859517) [](ancestors = 213859517,213858940)

Reply:
>What do you get when there is no class or class?? 

Why is this interesting? It is not meaningful to query this API when HasReferenceTypeConstraint is false (there is no class or class?), but the API returns false in this case since something has to be returned.

> Is there a difference between class in NNT(true) or NNT(false) context?

As everywhere else the context affects whether we know something or not. I do not believe there is a value to sprinkle comments about that all over. For the consumer of the API it really shouldn't matter why things are known or not known. 

---
In reply to: [213860360](https://github.com/dotnet/roslyn/pull/29586#discussion_r213860360) [](ancestors = 213860360,213859517,213858940)

Reply:
Then consider asserting that `HasReferenceTypeConstraint` is true in the implementations.

---
In reply to: [213861946](https://github.com/dotnet/roslyn/pull/29586#discussion_r213861946) [](ancestors = 213861946,213860360,213859517,213858940)

Reply:
> Then consider asserting that HasReferenceTypeConstraint is true in the implementations.

I will document that this API returns false when HasReferenceTypeConstraint  is false

---
In reply to: [213865435](https://github.com/dotnet/roslyn/pull/29586#discussion_r213865435) [](ancestors = 213865435,213861946,213860360,213859517,213858940)